src/linear_algebra/std_basis.lean

```diff
@@ -44,12 +44,35 @@ def std_basis (i : ι) : φ i →ₗ[R] (Πi, φ i) := pi (diag i)
 lemma std_basis_apply (i : ι) (b : φ i) : std_basis R φ i b = update 0 i b :=
 by ext j; rw [std_basis, pi_apply, diag, update_apply]; refl
 
+lemma coe_std_basis (i : ι) : ⇑(std_basis R φ i) = pi.single i :=
+funext $ std_basis_apply R φ i
+
 @[simp] lemma std_basis_same (i : ι) (b : φ i) : std_basis R φ i b i = b :=
 by rw [std_basis_apply, update_same]
 
 lemma std_basis_ne (i j : ι) (h : j ≠ i) (b : φ i) : std_basis R φ i b j = 0 :=
 by rw [std_basis_apply, update_noteq h]; refl
 
+section ext
+
+variables {R φ} {M : Type*} [fintype ι] [add_comm_monoid M] [semimodule R M]
+  {f g : (Π i, φ i) →ₗ[R] M}
+
+lemma pi_ext (h : ∀ i x, f (pi.single i x) = g (pi.single i x)) :
+  f = g :=
+to_add_monoid_hom_injective $ add_monoid_hom.functions_ext _ _ _ h
+
+lemma pi_ext_iff : f = g ↔ ∀ i x, f (pi.single i x) = g (pi.single i x) :=
+⟨λ h i x, h ▸ rfl, pi_ext⟩
+
+@[ext] lemma pi_ext' (h : ∀ i, f.comp (std_basis R φ i) = g.comp (std_basis R φ i)) : f = g :=
```

Comment:
As above, a `note [partially-applied ext lemmas]` would be good.

Can you add `pi_ext'_iff` for consistency?